Essentials/src/com/earth2me/essentials/commands/Commandtpa.java

```diff
@@ -35,13 +35,24 @@ public void run(Server server, User user, String commandLabel, String[] args) th
         }
         // Don't let sender request teleport twice to the same player.
         if (user.getConfigUUID().equals(player.getTeleportRequest()) && player.hasOutstandingTeleportRequest() // Check timeout
-            && player.isTpRequestHere() == false) { // Make sure the last teleport request was actually tpa and not tpahere
+            && !player.isTpRequestHere()) { // Make sure the last teleport request was actually tpa and not tpahere
             throw new Exception(tl("requestSentAlready", player.getDisplayName()));
         }
+
+        final Teleport teleport = player.getTeleport();
+        teleport.setTpType(Teleport.TeleportType.TPA);
+
+        // See if the target is on cooldown.
+        try {
+            teleport.cooldown(true);
+        } catch (Exception ignored) {
+            // The target is on cooldown
+            user.sendMessage(tl("requestDeniedCooldown", player.getDisplayName()));
```

Comment:
This will result in an incorrect message. If I /tpa someoneelse and I am on cooldown, the message will state that someoneelse is on cooldown, which is incorrect.

Reply:
@Ichbinjoe I've just looked over this PR. As far as I can tell, this PR is designed to prevent a `/tpa` if the target is on cooldown, and this is doing what it should? It has nothing to do with the requester.

Reply:
@md678685 line 43 (old) says user.getTeleport() however line 43(new) says player.getTeleport(). Thats the basis I've determined for my comment (which is now >6months old)(